src/lib/utils.ts

```diff
@@ -3,55 +3,63 @@
 
 import Log from "./logger";
 
-/** Test if a string is ascii or not */
-export const isAscii = /^[\s\x20-\x7E]*$/;
+type Base64ConvertibleValue = string | Buffer;
 
-/**
- * Encode all ascii values in a map to base64
- * @param obj The object to encode
- * @param skip A list of keys to skip encoding
- */
-export function convertToBase64Map(obj: { data?: Record<string, string> }, skip: string[]) {
-  obj.data = obj.data ?? {};
-  for (const key in obj.data) {
-    const value = obj.data[key];
-    // Only encode ascii values
-    obj.data[key] = skip.includes(key) ? value : base64Encode(value);
+interface Base64Map {
+  data?: Record<string, Base64ConvertibleValue>;
+}
+
+/** Base64 encode all values in a map */
+export function convertToBase64Map(obj: Base64Map): Base64Map {
```

Comment:
so looking at the main code. we're not actually copying the object, only creating a place for the skip. Since we're getting rid of the skip completely, we can overwrite the data in the existing object and not return anything.

// here's what I think it should be, we could even get rid of the first line and replace it with obj?.data, but downstream it might expect that to be {} versus undefined.
```
export function convertToBase64Map(obj: { data?: Record<string, string> }) {
  obj.data = obj.data ?? {};
  for (const key in obj.data) {
    obj.data[key] =  base64Encode(obj.data[key]);
  }
}
```
and
```
export function convertFromBase64Map(obj: { data?: Record<string, string> }) {
  obj.data = obj.data ?? {};
  for (const key in obj.data) {
    if (obj.data[key] == undefined) {
      obj.data[key] = "";
    } else {
      const decoded = base64Decode(obj.data[key]);
        obj.data[key] = decoded;
    }     
  }
}
```